### PR TITLE
Add public token dashboard

### PR DESCRIPTION
**Pull request by jefmoura**
## Purpose
We need to generate tokens, so the users are able to share resources publicly.

## Approach
- Created a Token generator for TolaData, so we can generate and validate token
- Added a new public status (URL), so they can share the dashboard via public URL
- Added a new field to store the token to create the public URL
- Added UUID to dashboard
- Created a signal for Dashboard, so we can generate tokens automatically

### Furter Info
Related ticket: https://github.com/Humanitec/ActivityAPI/issues/326

### CODE REVIEW

File: tola/utils.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        if not constant_time_compare(
+                self._make_token_with_timestamp(
+                    instance, ts, flag), token):
+            return False
+
+        return True
+
+    def _make_token_with_timestamp(self, instance, timestamp, flag=None):
+        # timestamp is number of days since 2001-1-1.  Converted to
+        # base 36, this gives us a 3 digit string until about 2121
```
Comment (Menda) [inline tola/utils.py]: We need to create a ticket to change the implementation in the year 2121

Comment (jefmoura) [reply tola/utils.py]: I won't be alive, so I don't see any problem :sweat_smile:

File: tola/tests/test_utils.py [hunk 2]
```diff
@@ -0,0 +1,220 @@
+# -*- coding: utf-8 -*-
+from django.test import TestCase, tag
+from django.utils import six
+from django.utils.crypto import salted_hmac
+from django.utils.http import int_to_base36
+import factories
```
[the remark this answers is not in the document]
Comment (jefmoura) [reply tola/tests/test_utils.py]: Really? hahaha

File: workflow/models.py [hunk 3]
```diff
@@ -381,18 +381,22 @@ def save(self, *args, **kwargs):
...
+    PUBLIC_URL = 'url'
     PUBLIC_ALL = 'all'
 
     PUBLIC_CHOICES = (
         (PUBLIC_ORG, 'Organization'),
+        (PUBLIC_URL, 'URL'),
         (PUBLIC_ALL, 'All')
     )
 
+    dashboard_uuid = models.CharField(max_length=255, editable=False, verbose_name='Dashboard UUID', default=uuid.uuid4, unique=True)
```
[the remark this answers is not in the document]
Comment (jefmoura) [reply workflow/models.py]: The problem is that we already have data in the DB but I try to use it and let you know.

File: tola/tests/test_utils.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Menda) [reply tola/tests/test_utils.py]: Yes, really xD

**Comment by jefmoura** [comment]
Apparently, it's working wit `UUIDField` @Menda

**Pull request merged**